tests/filter.json

```diff
@@ -478,6 +490,30 @@
         [42]
       ]
     },
+    {
+      "name": "on primitive, selects nothing",
+      "selector": "$.a[?@ == 1]",
+      "document": {"a": 1},
+      "result": []
+    },
```

Comment:
> This could be improved since the `.a` isn't necessary.

It seems that would be the first test then with primitive top-level value. I had created #57 to propose in general adding tests for that.

> Also, I don't think having a filter here improves the test. It could be any selector.

But those are (at least in the specification) two separate cases which independently say that only arrays and objects are supported, so maybe it would be good to have separate tests?

Reply:
> so maybe it would be good to have separate tests

As noted in other comments, we _do_ have those tests.

Reply:
(I thought we had those tests for primitives but we only have tests for e.g. "index selector against object".)